taskchampion/lib/src/kv.rs

```diff
@@ -68,11 +71,28 @@ impl PassByValue for TCKV {
 /// } TCKVList;
 /// ```
 #[repr(C)]
+#[derive(Debug)]
 pub struct TCKVList {
-    len: libc::size_t,
+    pub len: libc::size_t,
     /// total size of items (internal use only)
-    _capacity: libc::size_t,
-    items: *mut TCKV,
+    pub _capacity: libc::size_t,
+    pub items: *mut TCKV,
+}
+
+impl From<TaskMap> for TCKVList {
+    fn from(taskmap: TaskMap) -> TCKVList {
+        let vec = taskmap
+            .iter()
+            .map(|(k, v)| {
+                let key = RustString::from(k.as_ref());
+                let value = RustString::from(v.as_ref());
+                TCKV::as_ctype((key, value))
+            })
+            .collect();
+        // SAFETY:
+        //  - caller will free this list
+        unsafe { TCKVList::return_val(vec) }
+    }
 }
```

Comment:
This appears to be unused.

In general, we should prefer to use the PassByValue or PassByPointer types, rather than From/Into, as those have clearer safety requirements. For example, the `From<TCKVList> for TaskMap` implementation implicitly assumes that the TCKVList will not be used anymore by the caller.